### PR TITLE
Provide commit details to distribution API for additional context

### PR DESCRIPTION
**Pull request by DennisPan**
With the additional context of the commit information, distribution notification will be able to show richer information

### CODE REVIEW

File: Tasks/AppCenterDistributeV1/appcenterdistribute.ts [hunk 1]
```diff
@@ -167,6 +167,23 @@ function publishRelease(apiServer: string, releaseUrl: string, releaseNotes: str
...
     };
 
+    const branchName = process.env['BUILD_SOURCEBRANCHNAME'];
+    const sourceVersion = process.env['BUILD_SOURCEVERSION'];
+
+    // Builds staretd by App Center has the commit message set
+    const commitMessage = process.env['LASTCOMMITMESSAGE'];
+
+    // Including these information for distribution notification to have additional context
+    if (branchName && sourceVersion && commitMessage) {
```
Comment (flegallo) [inline Tasks/AppCenterDistributeV1/appcenterdistribute.ts]: Could `commitMessage` be null or empty? If so, we should remove it from the if condition.

Comment (DennisPan) [reply Tasks/AppCenterDistributeV1/appcenterdistribute.ts]: Do you want to show that the rest of the information (branch name and commit hash) in the notification, when there is no commit message? If a customer uses this task in VSTS (without App Center), then commit message would be null, where branch name and source version are likely available

File: Tasks/AppCenterDistributeV1/appcenterdistribute.ts [hunk 2]
```diff
@@ -167,6 +167,23 @@ function publishRelease(apiServer: string, releaseUrl: string, releaseNotes: str
         ]
     };
 
+    const branchName = process.env['BUILD_SOURCEBRANCHNAME'];
+    const sourceVersion = process.env['BUILD_SOURCEVERSION'];
+
+    // Builds staretd by App Center has the commit message set
```
Comment (PirateRunscope3) [inline Tasks/AppCenterDistributeV1/appcenterdistribute.ts]: nit: started, have

File: Tasks/AppCenterDistributeV1/appcenterdistribute.ts (hunk 1 above)
Comment (DennisPan) [reply Tasks/AppCenterDistributeV1/appcenterdistribute.ts]: Okay - updated

File: Tasks/AppCenterDistributeV1/appcenterdistribute.ts (hunk 2 above)
Comment (DennisPan) [reply Tasks/AppCenterDistributeV1/appcenterdistribute.ts]: lol

File: Tasks/AppCenterDistributeV1/Tests/L0.ts [hunk 3]
```diff
@@ -166,5 +168,28 @@ describe('AppCenterDistribute L0 Suite', function () {
...
+    it('Positive path: publish commit info (excluding commit message)', (done: MochaDone) => {
+        this.timeout(2000);
+
+        let tp = path.join(__dirname, 'L0PublishCommitInfo_2.js');
+        let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+        assert(tr.succeeded, 'task should have succeeded');
+
+        done()
```
Comment (madhurig) [inline Tasks/AppCenterDistributeV1/Tests/L0.ts]: any reason to leave out ; after done()

Comment (DennisPan) [reply Tasks/AppCenterDistributeV1/Tests/L0.ts]: No reason - error from copy paste existing code (fixed those as well)

**Pull request merged**